shell/browser/native_window.h

```diff
@@ -437,6 +438,8 @@ class NativeWindow : public base::SupportsUserData,
 
   NativeWindow(const gin_helper::Dictionary& options, NativeWindow* parent);
 
+  virtual void OnTitleChanged() {}
```

Comment:
What about reusing `WidgetDelegate::SetTitleChangedCallback` https://source.chromium.org/chromium/chromium/src/+/main:ui/views/widget/widget_delegate.cc;l=205-207 ?

Reply:
@deepak1556 that was my initial attempt :smile_cat: but WidgetDelegate only  has room for one callback and it's already used by the root view.

Reply:
Wouldn't they end up in the same path even if we override,

Current usage - https://source.chromium.org/chromium/chromium/src/+/main:ui/views/widget/root_view.cc;l=325-326 `RootView::UpdateAccessibleName`

When override, `Widget::UpdateWindowTitle` would still call `RootView::UpdateAccessibleName` https://source.chromium.org/chromium/chromium/src/+/main:ui/views/widget/widget.cc;l=1420

Maybe there is another path I am missing.

Reply:
You're not missing anything; the accessible name _does_ get set either way. We _could_ claim `WidgetDelegate::title_changed_callback_` without consequence `137.0.7149.0`.

My concern is that if upstream either changes their callback and/or changes when they set the callback, our code could break in some nonobvious The DIY `NativeWindow::OnTitleChanged()` is a safeguard against that.